django/views/i18n.py

```diff
@@ -272,18 +234,20 @@ def get_javascript_catalog(locale, domain, packages):
     pdict = {}
     maxcnts = {}
     catalog = {}
-    for k, v in t.items():
-        if k == '':
+    trans_fallback_cat = trans._fallback._catalog if trans._fallback else {}
+    for key in itertools.chain(trans_cat.keys(), trans_fallback_cat.keys()):
+        if key == '' or key in catalog:
             continue
-        if isinstance(k, six.string_types):
-            catalog[k] = v
-        elif isinstance(k, tuple):
-            msgid = k[0]
-            cnt = k[1]
+        value = trans_cat.get(key, trans_fallback_cat.get(key, ''))
```

Comment:
One possible issue here? the key is not present on trans_cat and not present on trans_fallback_cat, you will end up with an empty string value, and you still add it to the catalog. Then on the actual javascript for gettext a key with empty value will be present and you will instead return the empty value (because typeof('') == 'undefined' won't be true so you never return msgid)